### PR TITLE
New advanced integration with new card fields

### PR DESCRIPTION
**Pull request by wsbrunson**
Two big changes
* the existing `advanced-integration` moved to `advanced-integration-legacy` (Old hosted card fields)
* The new CardFields example took over `advanced-integration`

In addition
* two test pages for advanced integration, one for purchase and one for vault
* added endpoints for the vault api in `paypal-api.js`

### CODE REVIEW

**Review by malikshahid2 — APPROVED** [review]

File: advanced-integration-legacy/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Advanced Integration Example
+
+## Instructions
+
+1. Rename `.env.example` to `.env` and update `CLIENT_ID` and `APP_SECRET`.
```
Comment (jshawl) [inline advanced-integration-legacy/README.md:5]: I don't think there is an `.env.example` in this directory (advanced-integration-legacy)

File: advanced-integration/paypal-api.js [hunk 2]
```diff
@@ -44,43 +42,88 @@ export async function capturePayment(orderId) {
...
+const handleResponse = async (response) => {
   if (response.status === 200 || response.status === 201) {
     return response.json();
   }
 
-  const errorMessage = await response.text();
-  throw new Error(errorMessage);
-}
+  const error = new Error (await response.text())
+  error.status = response.status
```
Comment (jshawl) [inline advanced-integration/paypal-api.js:127]: TIL 🆒

File: advanced-integration/server.js [hunk 3]
```diff
@@ -1,20 +1,44 @@
...
 
-// render checkout page with client id & unique client token
 app.get("/", async (req, res) => {
   const clientId = process.env.CLIENT_ID;
   try {
-    const clientToken = await paypal.generateClientToken();
-    res.render("checkout", { clientId, clientToken });
+    res.render("index", { clientId });
+  } catch (err) {
+    handleError(res, err);
```
Comment (jshawl) [inline advanced-integration/server.js:23]: Will `res.render` ever throw?

File: advanced-integration/server.js [hunk 4]
```diff
@@ -1,20 +1,44 @@
...
+  }
+});
+
+app.get("/purchase", async (req, res) => {
+  const clientId = process.env.CLIENT_ID;
```
Comment (jshawl) [inline advanced-integration/server.js:28]: option to add this to `const { PORT = 8888 } = process.env;` above

**Review by jshawl — APPROVED** [review]
Looks great! I did not yet run this locally but can if you want - just lmk. I also left a few non-blocking comments for where I think some error handling can be removed. I think it would be helpful to include a readme in `advanced-integration` and also a deprecation notice in `advance-integration-legacy` <- just food for thought! Otherwise, :shipit: !

**Comment by gregjopa** [comment]
This PR looks good! I do think we should keep it as a draft PR until the public docs are upgraded with the new card fields. Right now, the current advanced integration doc has a link to this "advanced integration" folder inside this repo and that will break if we merge this PR: https://developer.pa
Screen Shot 2023-06-27 at 9 48 29 AM
ypal.com/docs/checkout/advanced/integrate#link-generateclienttoken

**Comment by wsbrunson** [comment]
> This PR looks good! I do think we should keep it as a draft PR until the public docs are upgraded with the new card fields. Right now, the current advanced integration doc has a link to this "advanced integration" folder inside this repo and that will break if we merge this PR: https://developer.paypal.com/docs/checkout/advanced/integrate#link-generateclienttoken

agreed! I've looped in Lance Han from the devrel team, they are going to give this a look and spin it up. I'll work with them to possibly take over the merging of this PR to coincide with their docs changes

File: advanced-integration/server.js (hunk 3 above)
Comment (wsbrunson) [reply advanced-integration/server.js:23]: its possible. I think that try/catch was there for `paypa.generateClientToken` but its possible there could be an issue rendering `index.ejs` or finding the file if the filesystem changes

**Review by Sinleeseul — APPROVED** [review]

**Review by MALIKSHAHID786 — APPROVED** [review]

**Review by MALIKSHAHID786 — APPROVED** [review]

**Comment by gregjopa** [comment]
I have a draft PR for adding the new card-fields component here in #84

**Pull request closed without merging**